### PR TITLE
Move SourceOptions into infra.passes.options

### PR DESCRIPTION
**Pull request by shonfeder**
This is just a simple code organization change consistent with #2018 and the
ongoing work on #1174.

---

- [x] Ran `make fmt-fix` (or had formatting run automatically on all files edited)
- [x] Rebase onto `main` once #2020 is merged.

### CODE REVIEW

**Comment by shonfeder** [comment]
Note this PR is quite small, and reviewing should hold off until the pending PRs it branches off of are merged in and this branch is rebased.

**Review by thpani — APPROVED** [review]

**Comment by thpani** [comment]
> and reviewing should hold off until

Oh... well, too late :)

**Comment by shonfeder** [comment]
Sorry for tagging in reviews prematurely! Thanks for reviewing :D

**Pull request merged**
